lib/cn/ltok.h

```diff
@@ -0,0 +1,102 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#ifndef HSE_CN_LTOK_H
+#define HSE_CN_LTOK_H
+
+#include <hse_util/atomic.h>
+#include <hse_util/inttypes.h>
+#include <hse_util/spinlock.h>
+
+/* MTF_MOCK_DECL(ltok) */
+
+enum ltok_token {
+    ltok_spill,
+    ltok_kcomp,
+    ltok_kvcomp,
+    ltok_split,
+    ltok_join
+};
+
+struct ltok {
```

Comment:
It would be nice if there was a comment explaining why some of these are counters and while others are booleans.